### PR TITLE
Validates catkey on registration page.

### LINKED ISSUE #3749: (new registration form) Validate catkey format during form entry on the new Argo registration form
**NOTE** This issue depends on the new registration form.

When a user enters a catkey on the new Argo registration form, validate that it matches an allowable pattern (defined in the cocina model). This should follow the example of how the form handles source id validation. Users should not be able to submit the form with a catkey that is not valid.

Currently, submitting an invalid catkey will take the user to a 500 error page.

Must be a PR against PR #3684 or must be done separately after PR #3684 is merged -- either way.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #3749

## Why was this change made? 🤔
Per Andrew.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (including writing to shared file systems or interacting with other SDR APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡
Unit

⚡ ⚠ If this change updates the Argo UI, run all integration tests that use Argo and create a PR on the integration test repo to fix anything this change breaks. ⚡

### CODE REVIEW

**Comment by andrewjbtw** [comment]
This looks like it's functioning correctly in terms of not allowing registration but the UI is still causing issues.

Submitting an invalid catkey triggers the page to reload, which results in resetting the values of the dropdown menus to their default values. This would result in registering with the incorrect values if a user doesn't realize what happened.

The form UI also looks weird after the error:
Screen Shot 2022-06-21 at 4 47 41 PM

Would it be possible to mimic the UI for invalid source id and only highlight the box in red with an exclamation point?

Screen Shot 2022-06-21 at 4 47 02 PM

The hover text to go along with the exclamation point could read "Please enter a valid catkey"

**Comment by andrewjbtw** [comment]
To clarify, when there's an invalid source id, the form doesn't submit and the page doesn't reload. I would like to follow that approach for catkey (and barcode but that's a different ticket).

**Comment by justinlittman** [comment]
Oh, I didn't realize that there was client-side validation. I just mimicked the server-side validation in the form.

**Review by ndushay — APPROVED** [review]
One suggestion for your consideration

File: app/javascript/controllers/registration_controller.js [hunk 1]
```diff
@@ -39,13 +39,43 @@ export default class extends Controller {
...
+        fetch(`/registration/catkey?catkey=${currentCatkey}`)
+          .then(response => {
+            if (response.ok) {
+              return response.json();
+            }
+            throw new Error('Cannot check catkey. Try again later.');
+          })
+          .then((data) => {
+            if (data) {
+              this.setValidation(field, "Not found in catalog")
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply app/javascript/controllers/registration_controller.js:69]: The validation message appears with the catkey field, so I don't think this is necessary. (If you feel strongly, let me know and I will change.)

**Comment by andrewjbtw** [comment]
Results of a second round of testing.

This check is too strict:

- the form appears to be checking if a catkey is unique in SDR. We need to support registering multiple items to the same catkey, since this comes up often in serials and multi-volume works.

These checks are working correctly:

- the check for invalid catkeys is working
- the check for non-existent catkey is working too, which is great

For the error message, instead of "Please use the requested format" could it say "Please enter a valid catkey"?

**Comment by andrewjbtw** [comment]
This looks good now with handling multiple catkeys. As discussed on Slack, I'll ticket the error message separately as it's more complicated than I realized and will be better to evaluate on its own in terms of prioritization.

**Review by ndushay — APPROVED** [review]

**Pull request merged**